populating an interface{} field with an []interface{}

### The following program `sample.go` triggers an unexpected result

```go
package main

import "fmt"

type test struct {
    v interface{}
    s string
}

type T struct {
    name string
}

func main() {
    t := []test{
        {
            v: []interface{}{
                T{"hello"},
            },
            s: "world",
        },
    }
    fmt.Println(t)
}
```


### Expected result

```console
$ go run ./sample.go
[{[{hello}] world}]
```


### Got

```console
$ yaegi ./sample.go
./sample.go:18:5: panic
run: interface conversion: interface {} is []interface {}, not []interp.valueInterface
goroutine 1 [running]:
runtime/debug.Stack()
	/Users/marc/sdk/go1.17.2/src/runtime/debug/stack.go:24 +0x88
github.com/traefik/yaegi/interp.(*Interpreter).Execute.func1(0x14000559b30)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/program.go:131 +0x8c
panic({0x102da8b40, 0x14000561200})
	/Users/marc/sdk/go1.17.2/src/runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.runCfg.func1(0x14000162840, 0x1400055b320, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:193 +0x164
panic({0x102da8b40, 0x14000561200})
	/Users/marc/sdk/go1.17.2/src/runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.genValueInterfaceArray.func1(0x14000162840)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/value.go:293 +0x1dc
github.com/traefik/yaegi/interp.doComposite.func2(0x14000162840)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:2610 +0x208
github.com/traefik/yaegi/interp.runCfg(0x1400055b320, 0x14000162840, 0x140004c7e60, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:201 +0x268
github.com/traefik/yaegi/interp.(*Interpreter).run(0x1400018db00, 0x140004c7e60, 0x140001626e0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:120 +0x3d4
github.com/traefik/yaegi/interp.(*Interpreter).Execute(0x1400018db00, 0x14000560bd0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/program.go:157 +0x1e0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0x1400018db00, {0x1400017c380, 0xdf}, {0x16db1b718, 0x19}, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/interp.go:593 +0x98
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0x1400018db00, {0x16db1b718, 0x19})
	/Users/marc/go/src/github.com/traefik/yaegi/interp/interp.go:486 +0xe8
main.runFile(0x1400018db00, {0x16db1b718, 0x19}, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:149 +0xb0
main.run({0x1400013c010, 0x1, 0x1})
	/Users/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:112 +0xdd4
main.main()
	/Users/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/yaegi.go:144 +0x384
```


### Yaegi Version

v0.10.0

### Additional Notes

_No response_

--- a/_test/issue-1308.go
+++ b/_test/issue-1308.go
@@ -1,0 +1,27 @@
+package main
+
+import "fmt"
+
+type test struct {
+	v interface{}
+	s string
+}
+
+type T struct {
+	name string
+}
+
+func main() {
+	t := []test{
+		{
+			v: []interface{}{
+				T{"hello"},
+			},
+			s: "world",
+		},
+	}
+	fmt.Println(t)
+}
+
+// Output:
+// [{[{hello}] world}]

--- a/interp/run.go
+++ b/interp/run.go
@@ -6,7 +6,6 @@ import (
 	"errors"
 	"fmt"
 	"go/constant"
-	"log"
 	"reflect"
 	"regexp"
 	"strings"
@@ -2608,7 +2607,6 @@ func doComposite(n *node, hasType bool, keyed bool) {
 		a := reflect.New(typ.TypeOf()).Elem()
 		typ.mu.Unlock()
 		for i, v := range values {
-			log.Println(n.cfgErrorf("composite"), i, a.Field(i).Type())
 			a.Field(i).Set(v(f))
 		}
 		d := value(f)

--- a/interp/run.go
+++ b/interp/run.go
@@ -6,6 +6,7 @@ import (
 	"errors"
 	"fmt"
 	"go/constant"
+	"log"
 	"reflect"
 	"regexp"
 	"strings"
@@ -2588,7 +2589,7 @@ func doComposite(n *node, hasType bool, keyed bool) {
 			values[fieldIndex] = func(*frame) reflect.Value { return reflect.New(rft).Elem() }
 		case isFuncSrc(val.typ):
 			values[fieldIndex] = genValueAsFunctionWrapper(val)
-		case isArray(val.typ) && val.typ.val != nil && isInterfaceSrc(val.typ.val):
+		case isArray(val.typ) && val.typ.val != nil && isInterfaceSrc(val.typ.val) && !isEmptyInterface(val.typ.val):
 			values[fieldIndex] = genValueInterfaceArray(val)
 		case isInterfaceSrc(ft) && !isEmptyInterface(ft):
 			values[fieldIndex] = genValueInterface(val)
@@ -2607,6 +2608,7 @@ func doComposite(n *node, hasType bool, keyed bool) {
 		a := reflect.New(typ.TypeOf()).Elem()
 		typ.mu.Unlock()
 		for i, v := range values {
+			log.Println(n.cfgErrorf("composite"), i, a.Field(i).Type())
 			a.Field(i).Set(v(f))
 		}
 		d := value(f)

